### ISSUE TITLE
Lint rpm based package managers: dnf, yum, zypper

### ISSUE THREAD

**Issue by m-ildefons** [hadolint/hadolint#470]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

Hadolint is a great tool, but unfortunately only supports linting for Debian- and Alpine-like package managers.
It would be a great improvement to be able to lint `yum`, `dnf` and `zypper` commands like `apt` commands:
  * No update/upgrade/dist-upgrade.
  * Use `-y` for yum, dnf and `-n` for zypper or these commands may fail.
  * Clean caches, e.g. `yum clean all` should end an installation command.

### Expected behavior

Linter complains about wrong application of `yum`:
`/dev/stdin:2 DLXXXX Do not use yum update`
`/dev/stdin:2 DLXXXX Do not use yum without '-y' flag.`

### Actual behavior

No output. Linter pretends Dockerfile is fine.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.18.0-15-g6664056
```

**Dockerfile** (if relevant)

```Dockerfile
FROM centos:8
RUN yum update
```

**Additional environment details (OS, stack version, etc.)**
Linux 4.19.0-10-amd64 #1 SMP Debian 4.19.132-1 (2020-07-24) x86_64 GNU/Linux
Docker version 18.09.1, build 4c52b90

**Comment by code-noggin** [comment]
Why is `yum update` bad? I often have a base image with some core packages that need to be updated with security fixes.

**Comment by m-ildefons** [comment]
@code-noggin the same reasoning as with apt-get applies:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get
Updating core packages can fail, especially in unprivileged containers.

In addition to the reason from the docker docs, `yum update` breaks reproducibility of images.
There is the scenario, where you build an app now and it works fine even with the `yum update` command in the Dockerfile, but 6 months from now, you make a change and therefore rebuild the Docker image and suddenly the image no longer works because the `yum update` pulled a newer, incompatible version of some library or something.

Maybe consider using `yum update-to -y foobar-x.y.z` or `yum install -y foobar-x.y.z` to fix the security issues?
This should avoid all problems I can think of right now, as you would need to name the packages and versions you want to update to explicitly.

**Comment by zemanlx** [comment]
Also, most of Linux distro vendors are updating their base regularly so usually all you need to do is update your base image.

**Comment by pstatham** [comment]
I understand the logic that `yum update` prevents builds that can be reproduced. For my projects, its more important to regularly build images with the latest security patches. I run full regression tests, which catch most breakage due to new libraries. So I'll just add an `--ignore` to my hadolint command in my CI pipeline. Unless you can think of another to way to do this?

> most of Linux distro vendors are updating their base regularly 

```shell
docker run --user=0 --rm -it centos:centos8 yum check-update
CentOS-8 - AppStream                                                                                                                       4.2 MB/s | 5.8 MB     00:01
CentOS-8 - Base                                                                                                                            1.5 MB/s | 2.2 MB     00:01
CentOS-8 - Extras                                                                                                                           26 kB/s | 8.1 kB     00:00

bind-export-libs.x86_64                                                             32:9.11.13-6.el8_2.1                                                             BaseOS
ca-certificates.noarch                                                              2020.2.41-80.0.el8_2                                                             BaseOS
centos-gpg-keys.noarch                                                              8.2-2.2004.0.2.el8                                                               BaseOS
centos-release.x86_64                                                               8.2-2.2004.0.2.el8                                                               BaseOS
centos-repos.x86_64                                                                 8.2-2.2004.0.2.el8                                                               BaseOS
librepo.x86_64                                                                      1.11.0-3.el8_2                                                                   BaseOS
zlib.x86_64                                                                         1.2.11-16.el8_2                                                                  BaseOS
```
